app/src/main/java/space/taran/arknavigator/mvp/model/repo/index/ResourceId.kt

```diff
@@ -14,32 +14,10 @@ typealias ResourceId = Long
 // Calculating CRC-32 hash of a file takes about the
 // same time as reading the file from internal storage.
 
-fun computeId(size: Long, file: Path): ResourceId {
-    Log.d(RESOURCES_INDEX,
-        "calculating hash of $file (size is ${size / MEGABYTE} megabytes)")
-
-    val crc32 = CRC32()
-    //todo: synchronize access
-
-    val source = Files.newInputStream(file)
-    val buffer = ByteArray(BUFFER_CAPACITY)
-
-    var total = 0L
-    var read = source.read(buffer)
+private external fun computeIdNative(size: Long, file: String): Long;
 
-    while (read > 0) {
-        crc32.update(buffer, 0, read)
-
-        total += read.toLong()
-        read = source.read(buffer)
-    }
-    source.close()
-
-    Log.d(RESOURCES_INDEX, "$total bytes has been read")
-    if (total != size) throw AssertionError(
-        "File wasn't read to the end")
-
-    return crc32.value
+fun computeId(size: Long, file: Path): ResourceId {
+    return computeIdNative(size, file.toString());
 }
 
 private const val BUFFER_CAPACITY = 512 * KILOBYTE
```

Comment:
Unused constant

Reply:
Thank you for the comment; I have updated this unneeded line on the new commit.